dlt/_workspace/runtime.py

```diff
@@ -0,0 +1,191 @@
+from dlt.common.runtime.run_context import active
+
+
+@dataclass
+class AuthInfo:
+    user_id: str
+    email: str
+    jwt_token: str
+
+
+class RuntimeAuthService:
+    """
+    Implements login, logout and auth check internals
+
+    Authentication is performed based on the JWT token stored in the global secrets. On top of that,
+    authorization uses organisation and workspace id stored in the local config. For that, depending on the usage,
+    either workspace run context or base run context is required.
+    """
+
+    auth_info: Optional[AuthInfo] = None
+
+    _run_context: RunContextBase
+    _local_workspace_id: Optional[str] = None
+    _remote_workspace_id: Optional[str] = None
+
+    def __init__(self, run_context: RunContextBase):
+        self._run_context = run_context
+
+    @property
+    def workspace_run_context(self) -> WorkspaceRunContext:
+        if isinstance(self._run_context, WorkspaceRunContext):
+            return self._run_context
+        else:
+            raise WorkspaceRunContextNotAvailable(self._run_context.run_dir)
+
+    @property
+    def run_context(self) -> RunContextBase:
+        return self._run_context
+
+    @property
+    def workspace_id(self) -> str:
+        if not self._remote_workspace_id or self._remote_workspace_id != self._local_workspace_id:
+            raise RuntimeOperationNotAuthorized()
+        return self._remote_workspace_id
+
+    def authenticate(self) -> AuthInfo:
+        self._read_token()
+        return self.auth_info
+
+    def login(self, token: str) -> AuthInfo:
+        self._save_token(token)
+        return self.auth_info
+
+    def logout(self) -> None:
+        self._delete_token()
+        self._remote_workspace_id = None
+
+    def authorize(self) -> str:
+        # Currently, ensuring workspace id is the same as default workspace id of the user
+        if not self._remote_workspace_id:
+            client = get_api_client(self)
+            me_response = me.sync(client=client)
+
+            if isinstance(me_response, MeResponse):
+                self._remote_workspace_id = str(me_response.default_workspace.id)
+            else:
+                raise RuntimeError("Failed to get me response")
+
+        config = resolve_configuration(WorkspaceRuntimeConfiguration())
+        self._local_workspace_id = config.workspace_id
+
+        if not self._local_workspace_id:
+            raise LocalWorkspaceIdNotSet(self._remote_workspace_id)
+        elif self._local_workspace_id != self._remote_workspace_id:
+            raise WorkspaceIdMismatch(self._local_workspace_id, self._remote_workspace_id)
+
+        return self.workspace_id
+
+    def overwrite_local_workspace_id(self) -> None:
+        local_toml_config = ConfigTomlProvider(self.workspace_run_context.settings_dir)
+        local_toml_config.set_value(
+            "workspace_id",
+            str(self._remote_workspace_id),
+            None,
+            WorkspaceRuntimeConfiguration.__section__,
```

Comment:
I'd go for known_sections.WORKSPACE, RuntimeConfiguration.__section__ to have it nicely scoped